backend/src/main/java/org/ukma/spring/crooodle/service/impl/RoomServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+
+    private final RoomRepository roomRepository;
+    private final HotelRepository hotelRepository;
+
+    @Override
+    public int create(Long hotelId, RoomCreateRequestDto requestDto) {
+        HotelEntity hotel = hotelRepository.findById(hotelId)
+            .orElseThrow(() -> new IllegalArgumentException("Hotel not found"));
+
+        RoomEntity room = RoomEntity.builder()
+            .name(requestDto.getName())
+            .capacity(requestDto.getCapacity())
+            .pricePerNight(requestDto.getPricePerNight())
+            .description(requestDto.getDescription())
+            .available(requestDto.isAvailable())
+            .hotel(hotel)
+            .build();
+
+        room = roomRepository.save(room);
+        return room.getId().intValue();
+    }
+
+    @Override
+    public RoomAdminResponseDto getAdmin(Long hotelId, Long roomId) {
+        RoomEntity room = roomRepository.findByIdAndHotelId(roomId, hotelId)
+            .orElseThrow(() -> new IllegalArgumentException("Room not found in the given hotel"));
+        return mapToAdminResponseDto(room);
+    }
+
+    @Override
+    public void edit(Long hotelId, Long roomId, RoomEditRequestDto requestDto) {
+        RoomEntity room = roomRepository.findByIdAndHotelId(roomId, hotelId)
+            .orElseThrow(() -> new IllegalArgumentException("Room not found in the given hotel"));
+
+        room.setName(requestDto.getName());
+        room.setCapacity(requestDto.getCapacity());
+        room.setPricePerNight(requestDto.getPricePerNight());
+        room.setDescription(requestDto.getDescription());
+        room.setAvailable(requestDto.isAvailable());
+
+        roomRepository.save(room);
+    }
+
+    @Override
+    public void delete(Long hotelId, Long roomId) {
+        RoomEntity room = roomRepository.findByIdAndHotelId(roomId, hotelId)
+            .orElseThrow(() -> new IllegalArgumentException("Room not found in the given hotel"));
+
+        roomRepository.delete(room);
+    }
+
+    @Override
+    public PageResponseDto<RoomAdminResponseDto> listAdmin(Long hotelId, RoomCriteriaDto criteriaDto) {
+        List<RoomEntity> rooms = roomRepository.findByHotelId(hotelId);
+
+        List<RoomAdminResponseDto> roomDtos = rooms.stream()
+            .map(this::mapToAdminResponseDto)
+            .collect(Collectors.toList());
+
+        return PageResponseDto.<RoomAdminResponseDto>builder()
+            .total(roomDtos.size())
+            .items(roomDtos)
+            .build();
+    }
+
+    @Override
+    public RoomResponseDto get(Long hotelId, Long roomId) {
+        RoomEntity room = roomRepository.findByIdAndHotelId(roomId, hotelId)
+            .orElseThrow(() -> new IllegalArgumentException("Room not found in the given hotel"));
+        return mapToResponseDto(room);
+    }
+
+    @Override
+    public PageResponseDto<RoomResponseDto> list(Long hotelId, RoomCriteriaDto criteriaDto) {
+        List<RoomEntity> rooms = roomRepository.findByHotelId(hotelId);
```

Comment:
Does not use criteria in any way? Is this your intended behaviour? Than at least put TODO